engine/schema/src/main/java/com/cloud/upgrade/DatabaseUpgradeChecker.java

```diff
@@ -486,6 +488,39 @@ public void check() {
         }
     }
 
+    private void initDistributedLock() {
+        LOGGER.info("Setting up distributed lock table if not created.");
+        TransactionLegacy txn = TransactionLegacy.open("initDistributedLock");
+        txn.start();
+        String errorMessage = "Unable to get the database connections";
+        try {
+            Connection conn = txn.getConnection();
+            errorMessage = "Unable to create distributed_lock table in the 'cloud' database ";
+            String sql = "CREATE TABLE IF NOT EXISTS `cloud`.`distributed_lock` (" +
```

Comment:
should we create this in code, and if so, shouldn't we also delete/truncate in code?

Reply:
I meant removing any of those locks, but in hindsight I think that is either handled elsewhere or might be even destructive (i.e. deleting a lock for another MS which is still running)